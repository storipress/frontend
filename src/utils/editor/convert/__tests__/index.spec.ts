import { config, htmlToOutputSpec, purify } from '..'

describe('embed-editor', (): void => {
  describe('htmlToOutputSpec', (): void => {
    it.skip('convert html to prosemirror output spec', (): void => {
      expect(htmlToOutputSpec('<p><strong>a</strong></p>')).toEqual([['p', {}, ['strong', {}, 'a']]])
    })
  })

  describe('purify', () => {
    let old = {}
    beforeEach(() => {
      old = config({})
    })
    afterEach(() => {
      config(old)
    })

    describe('when sanitize == false', () => {
      it.skip('purify remove script', () => {
        const html = purify('<div class="some-class"></div><script></script>').innerHTML
        expect(html).toContain('div')
        expect(html).toContain('some-class')
        expect(html).not.toContain('script')
        expect(html).toMatchSnapshot()
      })

      it.skip('purify remove iframe', () => {
        const html = purify('<div class="some-class"></div><iframe></iframe>').innerHTML
        expect(html).toContain('div')
        expect(html).toContain('some-class')
        expect(html).not.toContain('iframe')
        expect(html).toMatchSnapshot()
      })

      it.skip('purify could allow iframe', () => {
        const html = purify('<div class="some-class"></div><iframe></iframe>', true).innerHTML
        expect(html).toContain('div')
        expect(html).toContain('some-class')
        expect(html).toContain('iframe')
        expect(html).toMatchSnapshot()
      })
    })

    describe('when stanitize == true', () => {
      beforeEach(() => {
        config({ sanitize: false })
      })

      it('purify allow everything', () => {
        const html = purify('<div class="some-class"></div><iframe></iframe><script></script>').innerHTML
        expect(html).toContain('div')
        expect(html).toContain('some-class')
        expect(html).toContain('iframe')
        expect(html).toContain('script')
        expect(html).toMatchSnapshot()
      })
    })
  })
})
