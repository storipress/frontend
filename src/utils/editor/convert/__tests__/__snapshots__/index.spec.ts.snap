// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`embed-editor > purify > when stanitize == true > purify allow everything 1`] = `"<div class="some-class"></div><iframe></iframe><script></script>"`;
